src/org/sosy_lab/common/collect/SkipList.java

```diff
@@ -0,0 +1,756 @@
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.sosy_lab.common.collect;
+
+import com.google.common.base.Preconditions;
+import com.google.errorprone.annotations.Var;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.NoSuchElementException;
+import java.util.Random;
+import java.util.Set;
+import java.util.SortedSet;
+import javax.annotation.Nullable;
+
+/**
+ * A skip list implementation. Similar to ordered trees, but in list form. Achieves time complexity
+ * similar to ordered trees using a probabilistic approach.
+ *
+ * <p>Time Complexity:
+ *
+ * <ul>
+ *   <li>Insertion O(log n)
+ *   <li>Removal O(log n)
+ *   <li>Contains O(log n)
+ *   <li>Random-access O(n)
+ *   <li>Iteration O(n)
+ * </ul>
+ *
+ * <p>Based on William Pugh: Skip Lists. A Probabilistic Alternative to Balanced Trees, 1990.
+ *
+ * <p>By design, a skip list is a single linked list. To be able to implement a random-access add
+ * operation in a simple manner, we extended the skip list to be double linked.
+ *
+ * @param <T> type of the elements of this list
+ * @see java.util.concurrent.ConcurrentSkipListMap
+ */
+public class SkipList<T> implements OrderStatisticSet<T>, Serializable {
+
+  private static final long serialVersionUID = 1001345121457565238L;
+
+  private static final int LEVEL_ONE = 0;
+
+  private static class Node<T> implements Serializable {
+
+    private static final long serialVersionUID = 10045121457565238L;
+
+    // These lists are transient because they are manually serialized to prevent stack overflows due
+    // to recursive structure
+    private transient List<Node<T>> next;
+    private transient List<Node<T>> prev;
+    @Nullable private T value; // May be null, so be careful with serializing
+    private final int maxLvl;
+
+    private List<Integer> inBetweenCount;
+
+    private Node(@Nullable T pValue, int pLevel) {
+      value = pValue;
+      next = new ArrayList<>(Collections.nCopies(pLevel + 1, null));
+      prev = new ArrayList<>(Collections.nCopies(pLevel + 1, null));
+      inBetweenCount = new ArrayList<>(Collections.nCopies(pLevel + 1, 0));
+      maxLvl = pLevel;
+    }
+
+    @Nullable
```

Comment:
Is this formatted with `google-java-format`? I prefer annotations for return types on the same line as the return type.

Reply:
Yes, unfortunately.

> I  prefer annotations for return types on the same line as the return type.
Me too. The way it is formatted, it appears to be an annotation for the whole method.
Is there an easy way to change this?

Reply:
No, we can't change this.

But I noticed that you have many methods that are package-private but could be `private`. If you add the `private` modifier before the `@Nullable`, google-java-format will format as desired.